nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/launcher/connect/ConnectBroker.java

```diff
@@ -877,6 +936,13 @@ public LocalPackage pkgInstall(String pkgId) {
         return pkgInstall(pkgId, false);
     }
 
+    /**
+     * @since 10.2
+     */
+    protected boolean isLauncherChanged() {
+        return "true".equals(env.getProperty(LAUNCHER_CHANGED_PROPERTY));
```

Comment:
wrong location?

Reply:
Possible solution is to move the method to either `NuxeoLauncher` or `ConfigurationGenerator`. However, it requires changes on exception handling—we cannot raise `LauncherRestartException` in Connect Broker anymore. We need to modify the exit value in Nuxeo Launcher as a post-command handling. It will be something like:

```java
if (!cmdOK) {
    errorValue = EXIT_CODE_ERROR;
}
if (isRestartRequired) {
    errorValue = EXIT_CODE_LAUNCHER_CHANGED;
}
```

Compared to the exiting logic `System.exit(128)`, your solution is too risky. I cannot guarantee that it doesn't break anything.